src/gui/transferlistmodel.cpp

```diff
@@ -275,12 +271,15 @@ bool TransferListModel::setData(const QModelIndex &index, const QVariant &value,
 
 void TransferListModel::addTorrent(BitTorrent::TorrentHandle *const torrent)
 {
-    if (!m_torrents.contains(torrent)) {
-        const int row = m_torrents.size();
-        beginInsertRows(QModelIndex(), row, row);
-        m_torrents << torrent;
-        endInsertRows();
-    }
+    if (m_torrentMap.contains(torrent))
```

Comment:
I would assert that m_torrentMap doesn't contain torrent.

Reply:
Are you sure? I have no idea why there is the `if()` in the first place or it is really needed.

Reply:
TransferListModel is designed so that it must be synchronized with the original model (session torrents). It tracks both add/delete signals so there should always be integrity, unless there are errors in the code. So we should assert on these conditions instead of silently ignore possible errors.

Reply:
ok, done.